interns-react-weekly-assignments/src/components/Home/Keshav/Week4/WeatherDisplay/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { WeatherDisplayWrapper } from './styles';
+
+const WeatherDisplay = () => {
+  const [cityName, setCityName] = useState('');
+  const [apiCode, setApiCode] = useState<number | string>(-1);
+  const [weatherDetails, setWeatherDetails] = useState({
+    city: '',
+    temperature: null,
+    latitude: null,
+    longitude: null,
+    pressure: null,
+    humidity: null,
+    country: '',
+    description: ''
+  });
+
+  const apiKey = process.env.REACT_APP_WEATHER_API_KEY;
+  const url = 'https://api.openweathermap.org/data/2.5/weather?q=' + cityName + '&appid=' + apiKey + '&units=metric';
+
+  const fetchTemperature = () => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data);
+        setApiCode(data.cod);
```

Comment:
you can convert your statusCode to number before saving to have apiCode as only number value like this
const [apiCode, setApiCode] = useState<number>(-1);